framework/src/modules/chain/rounds/delegates.js

```diff
@@ -145,50 +121,6 @@ class Delegates {
 		return truncDelegateList;
 	}
 
-	/**
-	 * Generates delegate list and checks if block generator public key matches delegate id.
-	 *
-	 * @param {block} block
-	 * @param {function} cb - Callback function
-	 * @returns {setImmediateCallback} cb, err
-	 * @todo Add description for the params
-	 */
-	async validateBlockSlot(block) {
-		const round = this.slots.calcRound(block.height);
-		const activeDelegates = await this.generateDelegateList(
```

Comment:
As `generateDelegateList` is not used anymore, all the following functions in this file can be removed:
```
generateDelegateList
updateDelegateListCache
clearDelegateListCache
getKeysSortByVote
getDelegatesFromPreviousRound
```

Reply:
Decided to remove after `undo` and `apply` are implemented as the `getOutsiders` function is still using it.

Reply:
Created issue for this and added to Yatki's parent issue: https://github.com/LiskHQ/lisk-sdk/issues/4168

Reply:
There is an issue for this already: https://github.com/LiskHQ/lisk-sdk/issues/3685